scheduler/filter/slots.go

```diff
@@ -0,0 +1,48 @@
+package filter
+
+import (
+	"errors"
+	"github.com/docker/swarm/cluster"
+	"github.com/docker/swarm/scheduler/node"
+	"strconv"
+)
+
+var (
+	// ErrNoNodeWithFreeSlotsAvailable is exported
+	ErrNoNodeWithFreeSlotsAvailable = errors.New("No node with enough open slots available in the cluster")
+)
+
+//SlotsFilter only schedules containers with open slots.
+type SlotsFilter struct {
+}
+
+// Name returns the name of the filter
+func (f *SlotsFilter) Name() string {
+	return "containerslots"
+}
+
+// Filter is exported
+func (f *SlotsFilter) Filter(_ *cluster.ContainerConfig, nodes []*node.Node, _ bool) ([]*node.Node, error) {
+	result := []*node.Node{}
+
+	for _, node := range nodes {
+
+		if slotsString, ok := node.Labels["containerslots"]; ok {
+			slots, err := strconv.Atoi(slotsString) //if err => cannot cast to int, so no candidate for us
```

Comment:
I would be okay with both approaches.... i don't think its too much logging, but the problem is: who reads the syslog?! i guess nearly nobody... 

The other approach sounds better too me, someone startet a docker daemon with a faulty label value, so it will never be healthy to start any containers on it. Will the user see, "why" it is unhealthy? There is some kind of error field in the info view, right?